src/AppBundle/Entity/Report/Document.php

```diff
@@ -140,13 +156,34 @@ public function getReport()
     }
 
     /**
-     * @param Report $report
+     * @param Report|null $report
      *
      * @return $this
      */
-    public function setReport(Report $report)
+    public function setReport(Report $report = null)
```

Comment:
connecting both gives an easier way of seeing ALL the documents attached to the report, regardless of when they were submitted.
So to get all documents you'd do $report->getDocuments() and to get what was attached when you'd loop through report->getReportSubmissions() calling getDocuments on each submission It shouldnt be a repition because each document stores both the report and submission id, its two different 1:n relationships. Id leave the documents attached to both and by removing the null we also ensure enforce the fact that each document MUST be attached to a report (hence its call in the constructor)